service/history/api/respondactivitytaskfailed/api_test.go

```diff
@@ -0,0 +1,549 @@
+	"go.temporal.io/server/common/primitives/timestamp"
+	"go.temporal.io/server/service/history/api"
+	"go.temporal.io/server/service/history/consts"
+	"go.temporal.io/server/service/history/shard"
+	"go.temporal.io/server/service/history/tests"
+	"go.temporal.io/server/service/history/workflow"
+	wcache "go.temporal.io/server/service/history/workflow/cache"
+)
+
+type (
+	workflowSuite struct {
+		suite.Suite
+		*require.Assertions
+
+		controller        *gomock.Controller
+		shardContext      *shard.MockContext
+		namespaceRegistry *namespace.MockRegistry
+
+		workflowCache              *wcache.MockCache
+		workflowConsistencyChecker api.WorkflowConsistencyChecker
+
+		currentContext      *workflow.MockContext
+		currentMutableState *workflow.MockMutableState
+
+		activityInfo *persistence.ActivityInfo
+	}
+)
+
+func TestWorkflowSuite(t *testing.T) {
+	s := new(workflowSuite)
+	suite.Run(t, s)
+}
+
+func (s *workflowSuite) SetupSuite() {
+}
+
+func (s *workflowSuite) TearDownSuite() {
+}
+
+func (s *workflowSuite) SetupTest() {
+	s.Assertions = require.New(s.T())
+	s.controller = gomock.NewController(s.T())
+}
+
+func (s *workflowSuite) TearDownTest() {
+	s.controller.Finish()
+}
+
+type UsecaseConfig struct {
+	request             *historyservice.RespondActivityTaskFailedRequest
+	attempt             int32
+	activityId          string
+	activityType        string
+	startedEventId      int64
+	scheduledEventId    int64
+	taskQueueId         string
+	isActivityActive    bool
+	isExecutionRunning  bool
+	expectRetryActivity bool
+	retryActivityError  error
+	retryActivityState  enums2.RetryState
+	namespaceId         namespace.ID
+	namespaceName       namespace.Name
+	wfType              *commonpb.WorkflowType
+	tokenVersion        int64
+	tokenAttempt        int32
+	isCacheStale        bool
+	includeHeartbeat    bool
+}
+
+func (s *workflowSuite) Test_NormalFlowShouldRescheduleActivity_UpdatesWorkflowExecutionAsActive() {
+	ctx := context.Background()
+	uc := newUseCase(UsecaseConfig{
+		attempt:             int32(1),
+		startedEventId:      int64(40),
+		scheduledEventId:    int64(42),
+		taskQueueId:         "some-task-queue",
+		expectRetryActivity: true,
+		isCacheStale:        false,
+		retryActivityState:  enums2.RETRY_STATE_IN_PROGRESS,
+	})
+	request := s.newRespondActivityTaskFailedRequest(uc)
+	s.setupStubs(uc)
+
+	s.expectTimerMetricsRecorded(uc, s.shardContext)
+	s.currentContext.EXPECT().UpdateWorkflowExecutionAsActive(ctx, s.shardContext).Return(nil)
+
+	_, err := Invoke(ctx, request, s.shardContext, s.workflowConsistencyChecker)
+	s.NoError(err)
+}
+
+func (s *workflowSuite) Test_WorkflowExecutionIsNotRunning_ReturnWorkflowNotRunningError() {
+	uc := newUseCase(UsecaseConfig{
+		attempt:            int32(1),
+		startedEventId:     int64(40),
+		scheduledEventId:   int64(42),
+		isExecutionRunning: false,
+	})
+	request := s.newRespondActivityTaskFailedRequest(uc)
+	s.setupStubs(uc)
+	_, err := Invoke(
```

Comment:
```suggestion

	_, err := Invoke(
```

I like how the other tests help to visually separate the setup from the execution and assertion with a newline - let's do that here, too?